relayer/connection-tx.go

```diff
@@ -104,13 +104,17 @@ func ExecuteConnectionStep(src, dst *Chain) (success, last, modified bool, err e
 	// is chosen or a new connection is created.
 	// This will perform either an OpenInit or OpenTry step and return
 	if src.PathEnd.ConnectionID == "" || dst.PathEnd.ConnectionID == "" {
-		// TODO: Query for existing identifier and fill config, if possible
-		success, modified, err := InitializeConnection(src, dst, srcUpdateHeader, dstUpdateHeader, sh)
-		if err != nil {
-			return false, false, false, err
+		connectionID, found := FindMatchingConnection(src, dst)
```

Comment:
This should be called inside of `InitializeConnection`. It should only be called when we explicitly check that some identifier is missing ie
- src.ConnID is blank and dst.ConnID is blank (check for src)
- src.ConnID is blank and dst.Conn is not blank  (check for src)
- src.Conn is not blank and dst.Conn is blank (check for dst)